More intermitent issues on integration tests

The goal of this issue is to keep gathering problems seen on integration tests to find fixes and make them more resillient.

After https://github.com/kubearchive/kubearchive/issues/740, most of them were fixed but I've seen at least one more error on the installation phase:

- [Logs](https://github.com/kubearchive/kubearchive/actions/runs/12826696209/job/35767135850?pr=751) - Error when killing jobs because the process isn't found:

```
+ trap 'kill $(jobs -p)' EXIT
++ jobs -p
+ kill 19727
hack/kubearchive-install.sh: line 1: kill: (19727) - No such process
Error: Process completed with exit code 1.
```

- [Logs](https://github.com/kubearchive/kubearchive/actions/runs/12826448053/job/35766371043?pr=736) - Logging retrieval fails because secret credentials weren't found

```
HTTP request: /apis/batch/v1/namespaces/test-hmypwirb/cronjobs/generate-log-1/log
HTTP body: {"message":"logging secret user or password unset"}
The HTTP status returned is not OK, 400 Bad Request
```

--- a/.github/workflows/go_test_integration.yml
+++ b/.github/workflows/go_test_integration.yml
@@ -57,4 +57,4 @@ jobs:
       - name: Test Go code
         run: |
           export KO_DOCKER_REPO="kind.local"
-          go test -count=1 -v ./test/integration -tags=integration -timeout 30m
+          go test -count=1 -v ./test/integration -tags=integration -timeout 60m

--- a/test/integration/integration_test.go
+++ b/test/integration/integration_test.go
@@ -364,7 +364,8 @@ func TestArchiveAndRead(t *testing.T) {
 	}
 
 	// Retrieve cronjob logs
-
+	// NOTE that this will retrieve the url of the latest pod, but it would be more useful to get the logs from the first
+	// pod because that way the waiting for the logs to appear on splunk should be shorter.
 	url = fmt.Sprintf("https://localhost:8081/apis/batch/v1/namespaces/%s/cronjobs/generate-log-1/log", namespaceName)
 	request, errRequest = http.NewRequest("GET", url, nil)
 	if errRequest != nil {
@@ -397,13 +398,13 @@ func TestArchiveAndRead(t *testing.T) {
 		if len(body) == 0 {
 			return errors.New("could not retrieve the cronjob pod log")
 		}
+		fmt.Println("successfully retrieved logs")
 		return nil
 	}, retry.Attempts(20))
 
 	if retryErr != nil {
 		t.Fatal(retryErr)
 	}
-
 }
 
 func TestPagination(t *testing.T) {
@@ -539,43 +540,47 @@ func TestPagination(t *testing.T) {
 	defer close(portForward)
 
 	url := fmt.Sprintf("https://localhost:8081/api/v1/namespaces/%s/pods", namespaceName)
+	var list *unstructured.UnstructuredList
+	var getUrlErr error
 	err = retry.Do(func() error {
-		list, err := getUrl(&clientHTTP, token.Status.Token, url)
-		if err != nil {
-			t.Fatal(err)
+		list, getUrlErr = getUrl(&clientHTTP, token.Status.Token, url)
+		if getUrlErr != nil {
+			t.Fatal(getUrlErr)
 		}
 		// We want to wait until everything is stored in the DB to avoid out of order inserts
 		if len(list.Items) >= 30 {
 			return nil
 		}
 		return errors.New("could not retrieve Pods from the API")
-	}, retry.Attempts(60), retry.MaxDelay(4*time.Second))
+	}, retry.Attempts(80), retry.MaxDelay(4*time.Second))
 
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	url = fmt.Sprintf("https://localhost:8081/api/v1/namespaces/%s/pods?limit=10", namespaceName)
-	list, err := getUrl(&clientHTTP, token.Status.Token, url)
-	if err != nil {
-		t.Fatal(err)
+	initList, getUrlErrInitList := getUrl(&clientHTTP, token.Status.Token, url)
+	if getUrlErrInitList != nil {
+		t.Fatal(getUrlErrInitList)
 	}
-	assert.Equal(t, 10, len(list.Items))
+	assert.Equal(t, 10, len(initList.Items))
 
-	url = fmt.Sprintf("https://localhost:8081/api/v1/namespaces/%s/pods?limit=10&continue=%s", namespaceName, list.GetContinue())
-	continueList, err := getUrl(&clientHTTP, token.Status.Token, url)
-	if err != nil {
-		t.Fatal(err)
+	url = fmt.Sprintf("https://localhost:8081/api/v1/namespaces/%s/pods?limit=10&continue=%s", namespaceName, initList.GetContinue())
+	continueList, getUrlErrContinueList := getUrl(&clientHTTP, token.Status.Token, url)
+	if getUrlErrContinueList != nil {
+		t.Fatal(getUrlErrContinueList)
 	}
+	assert.Equal(t, 10, len(continueList.Items))
 
 	url = fmt.Sprintf("https://localhost:8081/api/v1/namespaces/%s/pods?limit=20", namespaceName)
-	allList, err := getUrl(&clientHTTP, token.Status.Token, url)
-	if err != nil {
-		t.Fatal(err)
+	allList, getUrlErrAllList := getUrl(&clientHTTP, token.Status.Token, url)
+	if getUrlErrAllList != nil {
+		t.Fatal(getUrlErrAllList)
 	}
+	assert.Equal(t, 20, len(allList.Items))
 
 	var listNames []string
-	for _, item := range list.Items {
+	for _, item := range initList.Items {
 		listNames = append(listNames, item.GetName())
 	}
 
@@ -623,6 +628,6 @@ func getUrl(client *http.Client, token string, url string) (*unstructured.Unstru
 		fmt.Printf("Couldn't unmarshal JSON, %s\n", err)
 		return nil, err
 	}
-	fmt.Printf("The HTTP status returned is OK, %s - %s \n", response.Status, string(body))
+	fmt.Printf("The HTTP status returned is OK, %s, returned %d items\n", response.Status, len(data.Items))
 	return &data, nil
 }

